optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -147,10 +148,103 @@
      * }
      */
 
-    // TODO
-    // join(..., planningVariableContainsCached(Talk::getPeriod, (Period a, Period b) -> a.overlaps(b)))
-    // get the period value range, does a cartesian product on it, so it maps every period to an overlapping periodList
-    // then keep an index from every period to all talks in an overlapping period (possible the same period)
+    /**
+     * For pairs of intervals [a,b) and [c,d), returns only those
+     * pairs where [a,b) and [c,d) overlap (or, in other
+     * words, there exists an x such that x in [a,b) and
+     * x in [c,d)). These are exactly the pairs where
+     * a < d and b > c (or in other words, when the first
+     * interval starts before the second one ends and the
+     * first interval ends after the second interval begins).
+     * 
+     * @param <A> The type of the first argument
+     * @param <B> The type of the second argument
+     * @param <Property_> The type used to define the interval, comparable
+     * 
+     * @param leftStartMapping Maps the first argument to the first interval start point
+     * @param leftEndMapping Maps the first argument to the first interval end point
+     * @param rightStartMapping Maps the second argument to the second interval start point
+     * @param rightEndMapping Maps the second argument to the second interval end point
+     * 
+     * @return An indexed joiner that filters the constraint stream to only include
+     *         elements (A,B) where A's and B's intervals (as defined by the function mapping)
+     *         overlap.
```

Comment:
We usually start with `never null, an indexed joiner...` And since it doesn't start with a capital letter, it doesn't end with a full stop.